deluge/core/core.py

```diff
@@ -999,6 +999,8 @@ def create_torrent(
         created_by=None,
         trackers=None,
         add_to_session=False,
+        include_v1=True,
+        include_v2=False,
```

Comment:
I don't feel it is a good idea to have these libtorrent implementation details leaking into the functions and API. It would be better to use something like a TorrentFormat Enum and have a `format` param that can accept str or Enum.

A tentative implementation attempt:

```
class TorrentFormat(str, Enum):
     V1 = 'v1_only'
     V2 = 'v2_only'
     HYBRID = 'hybrid'

     @classmethod
     def _missing_(cls, value):
         if not value:
             return cls.V1

         value = value.lower()
         for member in cls:
             if member.value.startswith(value):
                return member

     def to_lt_flag(self):
         if self.value == 'v1_only':
             return 64
         if self.value == 'v2_only':
             return 32
         return 0
```

This can be better represent available options to users.

Reply:
Did it a bit differently. The called it torrent_format everywhere to avoid clash with the builtin `format` function. The format names are `v1`, `v2` and `hybrid` (without the _only suffix). And enforce a valid format string in _missing_. Instead of returning `v1` on invalid format.